### DISCUSSION TITLE
Image Classification

### DISCUSSION THREAD

**Discussion by takdirzd** [teamalgoritma/community#388] [Q&A] [answered]
Misi kak mau tanya, ini pas implementasikan model ke data test seperti dibawah, katanya barisnya tidak sesuai antara baris data test, dan baris hasil prediksi.  Data testnya 294 baris, sedangkan hasil prediksinya hanya 264.

![image](https://user-images.githubusercontent.com/88231577/193796403-eb38fa0c-bac1-487a-b7ab-7ec8260d84f0.png)

kalau untuk splitting data untuk validasi datanya hanya 264 data/baris.
![image](https://user-images.githubusercontent.com/88231577/193796749-c0297c94-adab-4586-8fd0-5da3f07a3c1e.png)

**Comment by amaliasf** [comment, marked as answer]
Dear Pak @takdirzd 

Error tersebut didapatkan karena antara `df_test$id` dan `pred_test1` tidak memiliki panjang yang sama. Untuk itu bisa diverifikasi dengan melihat panjang dari data menggunakan `length()`. Pastikan panjangnya harus sama.

Kalau seharusnya sudah tepat pada validasi data hanya ada 264 baris, bisa dicek kembali bentuk data `df_test`, apakah ada step yang mengakibatkan bentuk data berubah. 

Semoga membantu Pak :)

**Reply by takdirzd** [reply]
Untuk ukuran data validationnya saya ubah/utak-atik di validation_split, sampai mendekati jumlah data asli yaitu 294, tetapi hanya 295, dan 293 data saja yang bisa.

![image](https://user-images.githubusercontent.com/88231577/194808941-ea829c2d-38e5-4d5d-925c-a47f004ca4c8.png)

untuk stepnya, di akhir saya hanya membaca csv nya trus langsung menerapkan modelnya, jadi data_testnya tidak melewati preprocessing yang membuat bentuk data berubah..

=============================================================================

tadinya juga saya manipulasi data aslinya jadi 295 baris saya mencopy data terakhir

![image](https://user-images.githubusercontent.com/88231577/194810015-377c9cec-2431-4ba8-a6e4-d7832a68e6b0.png)

akan tetapi pas di masukkan ke leaderboard, hasilnya malah error..

![image](https://user-images.githubusercontent.com/88231577/194809522-d629d726-7480-46a8-87e9-c06bd6e16b63.png)

**Reply by jafarhf1** [reply]
Dear @takdirzd,
Apakah bapak sudah mencoba untuk menambahkan parameter `row.names` saat membuat file submission seperti berikut ini ?
```
write.csv(submission, "submission-takdir.csv", row.names = F)
```
Apabila masih berkendala, boleh tolong diperlihatkan kode saat bapak membaca data dari folder tesnya sampai bisa menjadi variable `test_x` nya?

**Reply by takdirzd** [reply]
kalau sesuai download dari link dataset, harusnya file csv nya diluar folder test, disini sy hanya coba2 saja memasukkan/copy ke folder test nya, siapa tau bisa, tapi trnyata tidak.

![image](https://user-images.githubusercontent.com/88231577/195589619-92d0929e-8222-4aa1-ad30-286cf9ad584e.png)


masih error.. semua step guidance sudah saya ikuti disini https://algotech.netlify.app/blog/image-classification-cnn/
